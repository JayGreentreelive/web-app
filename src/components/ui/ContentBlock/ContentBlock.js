import React from 'react'
import classnames from 'classnames'
import { lowerCase, get } from 'lodash'
import { Block, Media } from '@christfellowshipchurch/web-ui-kit'

import ButtonRow from '../ButtonRow'

const titleClasses = classnames(
  'font-weight-bold'
)
const subtitleClasses = classnames(
  'pt-1'
)
const bodyClasses = classnames(
  'pt-1',
  'pb-4',
  'px-3'
)

const ContentBlock = ({
  contentLayout,
  images,
  videos,
  imageAlt,
  imageRatio,
  subtitle,
  title,
  htmlContent,
  callToAction,
  secondaryCallToAction
}) => (lowerCase(contentLayout) === 'background'
  ? (
    <Media
      imageUrl={get(images, '[0].sources[0].uri', '')}
      imageAlt={imageAlt}
      videoUrl={get(videos, '[0].sources[0].uri', '')}
      ratio={imageRatio}
    >
      <Block contentLayout="default" className="text-light">
        <Block.Title className='display-4'>
          {title}
        </Block.Title>

        <Block.Subtitle className=''>
          {subtitle}
        </Block.Subtitle>

        <Block.Body className={bodyClasses}>
          {htmlContent}
        </Block.Body>

        <ButtonRow callToAction={callToAction} secondaryCallToAction={secondaryCallToAction} />
      </Block>
    </Media>
  )
  : (
    <div className="container py-5">
      <div className="row">
        <Block
          layout={lowerCase(contentLayout)}
          media={get(images, '[0].sources[0].uri', null) || get(videos, '[0].sources[0].uri', null)
            ? {
              imageUrl: get(images, '[0].sources[0].uri', ''),
              imageAlt,
              videoUrl: get(videos, '[0].sources[0].uri', ''),
              ratio: imageRatio,
              showControls: true,
              rounded: true
            } : null}
        >

          <Block.Subtitle className={subtitleClasses}>
            {subtitle}
          </Block.Subtitle>

          <Block.Title className={titleClasses}>
            {title}
          </Block.Title>

          <Block.Body className={bodyClasses}>
            {htmlContent}
          </Block.Body>

          <ButtonRow callToAction={callToAction} secondaryCallToAction={secondaryCallToAction} />
        </Block>
      </div>
    </div>
  )
  )

export default ContentBlock