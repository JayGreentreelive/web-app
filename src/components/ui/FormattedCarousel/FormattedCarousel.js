import React from 'react'
import { get } from 'lodash'
import { Carousel, Block, Button } from '@christfellowshipchurch/web-ui-kit'
import '../../../styles.css'

const ButtonRow = ({ callToAction, secondaryCallToAction }) => (
  <div>
      {callToAction && callToAction.call !== '' && <Button type='primary' call={callToAction.call} />}
      <br></br>
      {secondaryCallToAction && secondaryCallToAction !== '' && <Button type='link' call={secondaryCallToAction.call} />}
  </div>
)

const FormattedCarousel = ({ children }) => (
  <div className="container pt-4">
    <div className="row py-5">
      <div className="col">
        <Carousel>
          {children.map(({
            __typename,
            images,
            imageAlt,
            videos,
            videoUrl,
            subtitle,
            title,
            htmlContent,
            callToAction,
            secondaryCallToAction
          }, i) => __typename === 'WebsiteBlockItem'
              ? (
                <>
                <h2 className='text-center text-dark pb-1'>{title}</h2>
                <Block
                  key={i}
                  layout={'default'}
                  media={{
                    imageUrl: get(images, '[0].sources[0].uri', ''),
                    imageAlt,
                    videoUrl: get(videos, '[0].sources[0].uri', ''),
                    ratio: '1by1',
                    circle: true,
                    className: 'carousel-img m-0 m-auto'
                  }} >
                  <div className='d-flex justify-content-center'>
                    <Block.Subtitle>{subtitle}</Block.Subtitle>
                    <Block.Body className='col-10 pt-3 font-weight-light'>{htmlContent}</Block.Body>
                  </div>
                  
                  {/* <ButtonRow callToAction={callToAction} secondaryCallToAction={secondaryCallToAction} /> */}

                </Block>
                </>
              ) : null
          )}
        </Carousel>
      </div>
    </div>
  </div>
)

export default FormattedCarousel