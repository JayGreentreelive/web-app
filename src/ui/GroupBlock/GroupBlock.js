import React from 'react'
import {
  useQuery
} from 'react-apollo'
import PropTypes from 'prop-types'
import {
  lowerCase, get, camelCase
} from 'lodash'
import {
  mapEdgesToNodes,
} from '../../utils'
import getGroupContentItems from '../../queries/getGroupContentItems'

import Loader from '../Loader'
import Block from '../Block'
import BackgroundContentBlock from '../BackgroundContentBlock'
import Carousel from '../Carousel'
import Accordion from '../Accordion'
import Tabs, { TabContent } from '../Tabs'
import { Feature } from '../../components/features'


const GroupBlock = ({
  id,
  groupLayout,
  accordionType
}) => {
  const { loading, error, data } = useQuery(getGroupContentItems, { variables: { id } })

  if (loading) return (
    <div className="vh-100 vw-100 d-flex justify-content-center align-items-center bg-light">
      <Loader />
    </div>
  )

  if (error) {
    console.error("ERROR: ", error)
    return <h1 className="text-center">There was an error loading block. Please try again.</h1>
  }

  const groupTitle = get(data, 'node.title', '')
  const groupBody = get(data, 'node.htmlContent', '')
  const blockItems = mapEdgesToNodes(data.node.childContentItemsConnection)

  if (!blockItems || !blockItems.length) return null

  switch (lowerCase(groupLayout)) {
    case 'row':
      return (
        <div className="row justify-content-center">
          {blockItems.map((n, i) => {
            switch (get(n, '__typename', '')) {
              case 'WebsiteBlockItem':
                if (camelCase(get(n, 'contentLayout', '')).includes('background')) {
                  return (
                    <BackgroundContentBlock
                      {...n}
                      className="col-md-6 d-flex align-items-center"
                      key={i} />
                  )
                } else {
                  return (
                    <Block
                      {...n}
                      className="col-md-6"
                      key={i} />
                  )
                }
              case 'WebsiteFeature':
                return (
                  <div key={i} className="col col-md-6 w-100 py-6 px-4">
                    <Feature name={get(n, 'feature', '')} />
                  </div>
                )
              default:
                return null
            }
          })}
        </div>
      )
    case 'accordion':
      const paginate = accordionType === 'paginate'
      return (
        <div className="row py-6">
          <div className="col-12 text-center mb-2">
            <div className="max-width-800">
              <h2>
                {groupTitle}
              </h2>
              {groupBody}
            </div>
          </div>
          <div className="col-12 max-width-1100">
            <Accordion paginate={paginate}>
              {blockItems.map((n, i) => {
                switch (get(n, '__typename', '')) {
                  case 'WebsiteBlockItem':
                    if (camelCase(get(n, 'contentLayout', '')).includes('background')) {
                      return (
                        <BackgroundContentBlock
                          {...n}
                          className="d-flex align-items-center"
                          key={i}
                        />
                      )
                    } else {
                      return (
                        <Block
                          {...n}
                          contentLayout="default"
                          key={i}
                        />
                      )
                    }
                  case 'WebsiteFeature':
                    return (
                      <div key={i} className="w-100 py-6 px-4">
                        <Feature name={get(n, 'feature', '')} />
                      </div>
                    )
                  default:
                    return null
                }
              })}
            </Accordion>
          </div>
        </div>
      )
    case 'carousel':
      return (
        <Carousel>
          {blockItems}
        </Carousel>
      )
    case 'tabs':
      return (
        <Tabs className="py-4">
          {blockItems.map((n, i) => <TabContent
            key={i}
            id={n.id}
            title={get(n, 'title', 'Click here')}
            icon={get(n, 'icon', null)}
          />)}
        </Tabs>
      )
    default:
      return null
  }
}

GroupBlock.propTypes = {
  groupLayout: PropTypes.oneOf([
    'row', 'accordion', 'carousel', 'tabs',
    'Row', 'Accordion', 'Carousel', 'Tabs',
    'none',
  ]),
  accordionType: PropTypes.oneOf(['default', 'paginate'])
}

GroupBlock.defaultProps = {
  groupLayout: 'none',
  accordionType: 'default'
}

export default GroupBlock