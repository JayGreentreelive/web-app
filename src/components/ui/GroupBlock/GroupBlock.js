import React from 'react'
import {
  useQuery
} from 'react-apollo'
import {
  lowerCase
} from 'lodash'
import {
  mapEdgesToNodes,
} from '../../../utils'
import getGroupContentItems from '../../../queries/getGroupContentItems'

import { Accordion, Row, Loader} from '@christfellowshipchurch/web-ui-kit'
import ContentBlock from '../ContentBlock'
import FormattedCarousel from '../FormattedCarousel'



const GroupBlock = ({ id, groupLayout }) => {
  const { loading, error, data } = useQuery(getGroupContentItems, { variables: { id } })

  if (loading) return (
    <div className="vh-100 vw-100 d-flex justify-content-center align-items-center bg-light">
      <Loader />
    </div>
  )
  if (error) {
    console.error("ERROR: ", error)
    return <h1 className="text-center">There was an error loading block. Please try again.</h1>
  }

  const groupTitle = data.node.title
  const groupBody = data.node.htmlContent
  const blockItems = mapEdgesToNodes(data.node.childContentItemsConnection)

  if (!blockItems || !blockItems.length) return null

  switch (lowerCase(groupLayout)) {
    case 'row':
      return (
        <Row>
          {blockItems.map((n, i) => n.__typename === 'WebsiteBlockItem'
            ? <ContentBlock {...n} key={i} /> : null)}
        </Row>
      )
    case 'accordion':
      return null
      // return (
      //   <Accordion
      //     blockTitle={groupTitle}
      //     blockBody={groupBody}
      //   >
      //     {blockItems.map((accordionItem, j) => {
      //       return (
      //         <div key={j} title={accordionItem.title}>
      //           <h2>{accordionItem.title}</h2>
      //           {accordionItem.htmlContent}
      //         </div>
      //       )
      //     }
      //     )}
      //   </Accordion>
      // )
    case 'carousel':
      return (
        <FormattedCarousel>
          {blockItems}
        </FormattedCarousel>
      )
    default:
      return null
  }
}

export default GroupBlock