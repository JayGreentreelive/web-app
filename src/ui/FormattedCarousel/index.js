import FormattedCarousel from './FormattedCarousel'

export default FormattedCarousel