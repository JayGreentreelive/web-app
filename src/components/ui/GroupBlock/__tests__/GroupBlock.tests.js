import React from 'react'
import renderer from "react-test-renderer"
import {Block, Media, Row} from '@christfellowshipchurch/web-ui-kit'
import ContentBlock from '../../ContentBlock'
import FormattedCarousel from '../../FormattedCarousel'
import GroupBlock from '../../GroupBlock'


describe("GroupBlock Renderer", () => {
  it("Displays a Row GroupBlock", () => {
    // const groupLayout = 'row'
    // expect(GroupBlock(groupLayout)).toBe('row')
    const tree = renderer.create(
      <Row>
        <ContentBlock/>
      </Row>  
    )
    expect(tree).toMatchSnapshot()
  })

  it("Displays a Carousel GroupBlock", () => {
    // const groupLayout = 'carousel'
    // expect(GroupBlock(groupLayout)).toBe('carousel')

    const children = [{
            images: '',
            imageAlt: '',
    }]

    const tree = renderer.create(
      <FormattedCarousel children={children} />
    )
    expect(tree).toMatchSnapshot()
  })
})
