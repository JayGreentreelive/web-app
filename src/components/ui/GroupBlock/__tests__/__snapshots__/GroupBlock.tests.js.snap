// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupBlock Renderer Displays a Carousel GroupBlock 1`] = `
<div
  className="container py-5"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-10 col-md-4"
    >
      <div
        className="carousel slide"
        data-ride="carousel"
      >
        <ol
          className="carousel-indicators"
        >
          <li
            className="active"
            onClick={[Function]}
          />
        </ol>
        <div
          className="carousel-inner container d-flex align-items-center pb-5"
        >
          <div
            className="item carousel-item active"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GroupBlock Renderer Displays a Row GroupBlock 1`] = `
<div
  className="col-12"
>
  <div
    className="container py-5"
  >
    <div
      className="row"
    >
      <div
        className="container-fluid "
      >
        <div
          className="row align-items-center"
        >
          <div
            className="col-12 col-md-6 order-first order-md-last"
          />
          <div
            className="col-12 col-md text-left"
          >
            <h6
              className="pt-1"
            />
            <h2
              className="font-weight-bold"
            />
            <div
              className="pt-1 pb-4 px-3"
            />
            <div>
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
